go/vt/mysqlctl/xtrabackupengine.go

```diff
@@ -554,16 +557,28 @@ func (be *XtrabackupEngine) extractFiles(ctx context.Context, logger logutil.Log
 		// Create the decompressor if needed.
 		if compressed {
 			var decompressor io.ReadCloser
-
+			var deCompressionEngine = bm.CompressionEngine
+			if deCompressionEngine == "" {
+				// For backward compatibility. Incase if Manifest is from N-1 binary
+				// then we assign the default value of compressionEngine.
+				deCompressionEngine = PgzipCompressor
+			}
 			if *ExternalDecompressorCmd != "" {
-				decompressor, err = newExternalDecompressor(ctx, *ExternalDecompressorCmd, reader, logger)
+				if deCompressionEngine == ExternalCompressor {
+					deCompressionEngine = *ExternalDecompressorCmd
+					decompressor, err = newExternalDecompressor(ctx, deCompressionEngine, reader, logger)
+				} else {
```

Comment:
If I'm reading this right it looks like if the user specifies `--external-decompressor`, but the manifest doesn't specify that the compression engine is `external`, then we'll ignore the user's request and instead decompress according to the manifest. Is that right?

Reply:
yes.

Reply:
If the user is coming from N-1 binary version, where they don't have the compression engine specified in their manifest, then how will they be able to decompress a backup that was compressed with non-default compression engine?

Reply:
This situation is not possible because in N-1 we didn't allow them to use anything other than the default (`pgzip`)